test/es-module/test-esm-cjs-load-error-note.mjs

```diff
@@ -0,0 +1,113 @@
+const Import4 = fixtures.path('/es-modules/es-note-unexpected-import-4.cjs');
+const Import5 = fixtures.path('/es-modules/es-note-unexpected-import-5.cjs');
+
+// Expect note to be included in the error output
+const expectedNote = 'Note: To load an ES module ' +
+'set "type": "module" in the package.json ' +
+'or use the .mjs extension.\n\n';
+
+const expectedCode = 1;
+const expectedSignal = null;
+
+
+const pExport1 = spawn(process.execPath, ['--experimental-modules', Export1]);
+let pExport1Stderr = '';
+pExport1.stderr.setEncoding('utf8');
+pExport1.stderr.on('data', (data) => {
+  pExport1Stderr += data;
+});
+pExport1.on('close', mustCall((code, signal) => {
+  assert.strictEqual(code, expectedCode);
+  assert.strictEqual(signal, expectedSignal);
+  assert.ok(pExport1Stderr.includes(expectedNote));
+}));
+
+const pExport2 = spawn(process.execPath, ['--experimental-modules', Export2]);
+let pExport2Stderr = '';
+pExport2.stderr.setEncoding('utf8');
+pExport2.stderr.on('data', (data) => {
+  pExport2Stderr += data;
+});
+pExport2.on('close', mustCall((code, signal) => {
+  assert.strictEqual(code, expectedCode);
+  assert.strictEqual(signal, expectedSignal);
+  assert.ok(pExport2Stderr.includes(expectedNote));
+}));
+
+
+const pImport1 = spawn(process.execPath, ['--experimental-modules', Import1]);
+let pImport1Stderr = '';
+pImport1.stderr.setEncoding('utf8');
+pImport1.stderr.on('data', (data) => {
+  pImport1Stderr += data;
+});
+pImport1.on('close', mustCall((code, signal) => {
+  assert.strictEqual(code, expectedCode);
+  assert.strictEqual(signal, expectedSignal);
+  assert.ok(pImport1Stderr.includes(expectedNote));
+}));
+
+// Note this test shouldn't include the note
+const pImport2 = spawn(process.execPath, ['--experimental-modules', Import2]);
+let pImport2Stderr = '';
+pImport2.stderr.setEncoding('utf8');
+pImport2.stderr.on('data', (data) => {
+  pImport2Stderr += data;
+});
+pImport2.on('close', mustCall((code, signal) => {
+  // As this exits normally we expect 0
+  assert.strictEqual(code, 0);
+  assert.strictEqual(signal, expectedSignal);
+  assert.ok(!pImport2Stderr.includes(expectedNote));
+}));
+
+const pImport3 = spawn(process.execPath, ['--experimental-modules', Import3]);
+let pImport3Stderr = '';
+pImport3.stderr.setEncoding('utf8');
+pImport3.stderr.on('data', (data) => {
+  pImport3Stderr += data;
+});
+pImport3.on('close', mustCall((code, signal) => {
+  assert.strictEqual(code, expectedCode);
+  assert.strictEqual(signal, expectedSignal);
+  assert.ok(pImport3Stderr.includes(expectedNote));
+}));
+
+
+const pImport4 = spawn(process.execPath, ['--experimental-modules', Import4]);
+let pImport4Stderr = '';
+pImport4.stderr.setEncoding('utf8');
+pImport4.stderr.on('data', (data) => {
+  pImport4Stderr += data;
+});
+pImport4.on('close', mustCall((code, signal) => {
+  assert.strictEqual(code, expectedCode);
+  assert.strictEqual(signal, expectedSignal);
+  assert.ok(pImport4Stderr.includes(expectedNote));
```

Comment:
This is the line that is failing on Windows in CI. It might help to provide a custom error message here and elsewhere that `assert.ok()` is being used:

```suggestion
  assert.ok(pImport4Stderr.includes(expectedNote),
            `${expectedNote} not found in ${pImport4Stderr}`);
```

Reply:
added additional test and wrote messages where assert is used.